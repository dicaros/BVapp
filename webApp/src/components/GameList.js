import React from 'react';
import { currentday, timedash, todaydash } from '../functions/functions'
import { url } from '../constants/constants'

var countpage = 0 + ' '

class GameList extends React.Component {

    // not used
/* 
   async handleChange(event) {
         await this.props.setSize(event.target.value);
         await this.props.setPage('prev', this.props.listurl, 1, this.props.items.page.totalPages)
   };
*/
 
/*   switchPage (direction) {
         this.props.setPage(direction, this.props.items.page.totalPages)
           };*/

async handleSelectGame(id) {
    await this.props.setGame(id)
    await this.props.fetchData('api/gameparticipantsget', url, '?id='+id)
    await this.props.fetchData('api/game2', url, '?id='+id)
    this.props.setNavigate('gamedetails')           
}

   async updateRecord(firstName, lastName, description, url) {  
    if (this.props.loginsuccess) {
       await  this.props.updateRecord(firstName, lastName, description, url)
         this.refresh();
      }
    }
 
   render() {      
     var tomorrow = new Date(Date.now()+1*24*60*60*1000);

         const TableHeader = () => {
                         return (
                                       <thead>
                                           <tr> 
                                              <th className = 'gamelist'>
                                              <div className='closewindowdiv'><span className = 'createbuttonspan'><button onClick={() => this.props.setNavigate('create')}>+ Create New</button></span>                                            
                                                 <br></br>Games near your location</div>
                                              </th>
                                           </tr>
                                       </thead>
                                 )
           };
 
         const Tablerow = () => {
            return(                    
                  this.props.items._embedded.games.map ((row, index) =>                   
                   {
                     if(true || row.gamedate > todaydash(currentday()) || (row.gamedate == todaydash(currentday()) && row.gametime >= timedash(currentday())))
                     {
                     return(
                             <div key={index} className='gameorganizer'>
                                  <li onClick={() => this.handleSelectGame(row.id)} 
                                      className='gamelist' key={index}>
                                        <span className='today'> 
                                            {(row.gamedate == todaydash(currentday())) && 'Today, '}
                                            {(row.gamedate == todaydash(currentday(tomorrow))) && 'Tomorrow, '}
                                            {(row.gamedate > todaydash(currentday(tomorrow))) && row.gamedate+ ', '}
                                            {String(row.gametime).substring(0, 5) + ' '}</span> 
                                           <br></br>
                                           @{(row._embedded.sportcenter.name) + ', '} 
                                            {row._embedded.sportcenter.street}
                                        <span className='gameorganizer'>Organized by: {row.myuser.name}</span>
                                  </li>
                            </div>
                         )
                      }
                    }
                  )
           )
         }

         const TableBody = () => {                       
                                    return (<tbody> 
                                                <tr>
                                                     <td className = 'gamelist' width='20%'>
                                                          <ul className='gamelist'><Tablerow/></ul>
                                                     </td>    
                                                </tr>                         
                                            </tbody>)
           }
/*                 const TableFooter = () => {
                                     return (
                                      countpage = this.props.items.page.totalPages + ' ',
                                       <tfoot>
                                         <tr id = 'trfooter'>
                                           <td>
                                                     <center>  <button onClick={() => this.switchPage('prev')}>Prev</button><span>&nbsp;&nbsp;</span>
                                                       <button onClick={() => this.switchPage('next')}>Next</button><span>&nbsp;&nbsp;</span>
                                  
                                                       {this.props.page+1}/{countpage}
                                                       
                                                        <select value={this.props.size} onChange={this.handleChange.bind(this)}>
                                                           <option value={5}>5</option>
                                                           <option value={10}>10</option>
                                                           <option value={20}>20</option>
                                                           <option value={50}>50</option>
                                                           <option value={100}>100</option>
                                                       </select>                                                   </center>
                                       </td>
                               
                                     </tr>
                                 </tfoot>
                              );   
                           }
  */      
          return (
                   <table className='tablegames'>
                       <TableHeader />
                       <TableBody  />
    
                   </table>
                 );
     }
 }
 
export default GameList;