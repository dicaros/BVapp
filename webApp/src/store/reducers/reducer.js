import { combineReducers } from 'redux';

export function navigate(state = 'games', action) {
    switch (action.type) {
        case 'SET_NAVIGATE':
            return action.navigate;        default:
            return state;
    }
}
export function page(state = 0, action) {
    switch (action.type) {
        case 'CURRENT_PAGE':
            return action.page;        default:
            return state;
    }
}
export function size(state = 1000000, action) {
    switch (action.type) {
        case 'CURRENT_SIZE':
            return action.size;        default:
            return state;
    }
}

export function currentgame(state = 0, action) {
    switch (action.type) {
        case 'CURRENT_GAME':
            return action.currentgame;        default:
            return state;
    }
}

export function isError(state = false, action) {
    switch (action.type) {
        case 'IS_ERROR':
            return action.hasErrored;        default:
            return state;
    }

}

export function isLoading(state = false, action) {
    switch (action.type) {
        case 'IS_LOADING':
            return action.isLoading;        default:
            return state;
    }
}

export function items(state = [], action) {
    switch (action.type) {
        case 'ITEMS_FETCH_DATA_SUCCESS':
            return action.items;        default:
            return state;
    }
}

export function loginsuccessfull(state = null, action) {
    switch (action.type) {
        case 'LOGIN_SUCCESS':
            return action.loginsuccessfull;        default:
            return state;
    }
}

export function nitems(state = 0, action) {
    switch (action.type) {
        case 'N_ITEMS':
            return action.nitems;        default:
            return state;
    }
}

export function gameparticipantsitems(state = [], action) {
    switch (action.type) {
        case 'GAMEPARTICIPANT_FETCH_DATA_SUCCESS':
            return action.gameparticipantsitems;        default:
            return state;
    }
}

export function sportcenteritems(state = [], action) {
    switch (action.type) {
        case 'SPORTCENTER_FETCH_DATA_SUCCESS':
            return action.sportcenteritems;        default:
            return state;
    }
}

export function useritems(state = [], action) {
    switch (action.type) {
        case 'USER_FETCH_DATA_SUCCESS':
            return action.useritems;        default:
            return state;
    }
}

export function registration(state = [], action) {
    switch (action.type) {
        case 'REGISTRATION_STATUS':
            return action.registration;        default:
            return state;
    }
}

export function joingame(state = [], action) {
    switch (action.type) {
        case 'JOINGAME_STATUS':
            return action.joingame;        default:
            return state;
    }
}

export function singlegameitems(state = [], action) {
    switch (action.type) {
        case 'SINGLEGAME_FETCH_DATA_SUCCESS':
            return action.singlegameitems;        default:
            return state;
    }
}

export function username(state = 'Guest', action) {
    switch (action.type) {
        case 'SET_USER':
            return action.username;        default:
            return state;
    }
}

export default combineReducers({
    currentgame,
    gameparticipantsitems,
    items,
    isError,
    isLoading,
    joingame,
    loginsuccessfull,
    navigate,
    nitems,
    page,
    registration,
    singlegameitems,
    size,
    sportcenteritems,
    useritems,
    username,
});