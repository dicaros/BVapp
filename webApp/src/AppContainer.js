import { getUser, loadData, newGame, newUser, setCurrentGame, setCurrentPage, setCurrentSize,
          setNavigate, signupGame, thelogin, thelogout, updateRecord } from './store/actions/actions';

import { connect } from 'react-redux';

import AppHeader from './components/AppHeader';
import GameList from './components/GameList';
import GameDetails from './components/GameDetails';
import UserDetails from './components/UserDetails';
import LoginPage from './components/LoginPage';
import RegisterUser from './components/RegisterUser';
import CreateGame from './components/CreateGame';
import MainContainer from './components/MainContainer';

const mapStateToProps = (state) => {
    return {
         gameparticipantsitems: state.gameparticipantsitems,
         isError: state.isError,
         isLoading: state.isLoading,
         loginsuccess: state.loginsuccessfull,
         listsportcenter: state.listsportcenter,
         items: state.items,
         joingame: state.joingame,
         nitems: state.nitems,
         navigate: state.navigate,
         page: state.page,
         size: state.size,
         singlegameitems: state.singlegameitems,         
         sportcenteritems: state.sportcenteritems,
         useritems: state.useritems,
         username: state.username,
         registration: state.registration,
         currentgame: state.currentgame
       };
  };
  
  const mapDispatchToProps = (dispatch) => {
  return {
      fetchData: (request, url, sort, page, size) => dispatch(loadData(request, url, sort, page, size)),
      doLogin: (loginurl, target) => dispatch(thelogin(loginurl, target)),                                    
      doLogout: (logouturl) => dispatch(thelogout(logouturl)),
      setNavigate: (navigate) => dispatch(setNavigate(navigate)),
      setGame: (id) => dispatch(setCurrentGame(id)),
      setSize: (size) => dispatch(setCurrentSize(size)),
      setPage: (direction, pagenum) => dispatch(setCurrentPage(direction, pagenum)),                              
      signupGame: (url, noshow, game_id) => dispatch(signupGame(url, noshow, game_id)),
      addNew: (sportcenterid, isPrivate, gameDate, gameTime, description, url, nitems, priceperperson, kurt) => dispatch(newGame(sportcenterid, isPrivate, gameDate, gameTime, description, url, nitems, priceperperson, kurt)),
      getUser: () => dispatch(getUser()),
      updateRecord: (firstName, lastName, description, url) => dispatch(updateRecord(firstName, lastName, description, url)),
      newUsers: (url, target) => dispatch(newUser(url, target)),
    };
  };

  export const CreateComponent = connect(mapStateToProps, mapDispatchToProps)(CreateGame);
  export const GameComponent = connect(mapStateToProps, mapDispatchToProps)(GameList);
  export const GameDetailsComponent = connect(mapStateToProps, mapDispatchToProps)(GameDetails);
  export const HeaderComponent = connect(mapStateToProps, mapDispatchToProps)(AppHeader);
  export const LoginComponent = connect(mapStateToProps, mapDispatchToProps)(LoginPage);
  export const MainComponent = connect(mapStateToProps, mapDispatchToProps)(MainContainer);
  export const RegisterComponent = connect(mapStateToProps, mapDispatchToProps)(RegisterUser);
  export const UserComponent = connect(mapStateToProps, mapDispatchToProps)(UserDetails);
