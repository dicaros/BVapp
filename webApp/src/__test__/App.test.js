// counter.test.js
import React from 'react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { initialState, reducer } from '../store/reducers/reducer.js'
import { CreateComponent, GameComponent, LoginComponent, UserComponent, GameDetailsComponent, UserProfileComponent, RegisterComponent } from '../AppContainer';
import GameList from '../components/GameList.js'

// this is a handy function that I normally make available for all my tests
// that deal with connected components.
// you can provide initialState for the entire store that the ui is rendered with
function renderWithRedux(
  ui,
  { initialState, store = createStore(reducer, initialState) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  }
}

test('can render with redux with defaults', () => {
  const { getByTestId, getByText } = renderWithRedux(<GameList />)
  fireEvent.click(getByText('+'))
  expect(getByTestId('count-value')).toHaveTextContent('1')
})
