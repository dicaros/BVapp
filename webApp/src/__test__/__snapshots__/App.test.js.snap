// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <div
      class="topheader"
    >
      <table
        class="tableheader"
      >
        <thead>
          <tr>
            <td
              class="headermenu_left"
            >
              <a
                href="/"
              >
                <img
                  alt="Logo"
                  class="logo"
                  src="{logo}"
                />
              </a>
            </td>
            <td
              class="headermenu_center"
            />
            <td
              class="headermenu_right"
            >
              <nav>
                <ul>
                  <li>
                    <img
                      alt="O"
                      class="userlogo"
                      src="UserIcon.png"
                    />
                  </li>
                  <li>
                    <div
                      class="container"
                    >
                      <div
                        class="bar1"
                      />
                      <div
                        class="bar2"
                      />
                      <div
                        class="bar3"
                      />
                    </div>
                    <ul
                      class="listmenuhidden"
                    >
                      <li>
                        <a
                          href="/"
                        >
                          Logout
                        </a>
                      </li>
                      <li>
                        <a
                          href="#"
                        >
                          Game list
                        </a>
                      </li>
                      <li>
                        <a
                          href="#"
                        >
                          Create game
                        </a>
                      </li>
                      <li>
                        <div
                          class="line"
                        />
                      </li>
                      <li>
                        <a
                          href="#"
                        />
                      </li>
                    </ul>
                  </li>
                </ul>
              </nav>
            </td>
          </tr>
        </thead>
      </table>
    </div>
  </header>
  <section
    class="appbody"
  >
    <center />
  </section>
</DocumentFragment>
`;
